frontend/src/component/playground/Playground/PlaygroundResultsTable/FeatureResultInfoPopoverCell/FeatureStrategyList/StrategyList/StrategyItem/StrategyExecution/SegmentExecution/SegmentExecution.tsx

```diff
@@ -1,76 +1,37 @@
-    color: theme.palette.error.main,
-    display: 'inline-flex',
-    justifyContent: 'center',
-    marginLeft: 'auto',
-    gap: theme.spacing(1),
-}));
+};
 
-export const SegmentExecution: VFC<ISegmentExecutionProps> = ({
-    segments,
+export const SegmentExecution: FC<SegmentExecutionProps> = ({
+    segment,
     input,
 }) => {
-    if (!segments) return null;
+    const constraintList =
+        segment.constraints.length > 0 ? (
+            <ConstraintsList>
+                {segment.constraints.map((constraint) => (
+                    <ConstraintExecution
+                        key={objectId(constraint)}
+                        constraint={constraint}
+                        input={input}
+                    />
+                ))}
+            </ConstraintsList>
+        ) : undefined;
 
     return (
-        <>
-            {segments.map((segment, index) => (
-                <Fragment key={segment.id}>
-                    <SegmentItem
-                        segment={segment}
-                        constraintList={
-                            <ConstraintExecution
-                                constraints={segment.constraints}
-                                input={input}
-                            />
-                        }
-                        headerContent={
-                            <ConditionallyRender
-                                condition={!segment.result}
-                                show={
-                                    <SegmentResultTextWrapper>
-                                        <Typography
-                                            variant={'subtitle2'}
-                                            sx={{ pt: 0.25 }}
-                                        >
-                                            segment is false
-                                        </Typography>
-                                        <span>
-                                            <CancelOutlined />
-                                        </span>
-                                    </SegmentResultTextWrapper>
-                                }
-                                elseShow={undefined}
-                            />
-                        }
-                        isExpanded
-                    />
-                    <ConditionallyRender
-                        condition={
-                            // Add IF there is a next segment
-                            index >= 0 &&
-                            segments.length > 1 &&
-                            // Don't add if it's the last segment item
-                            index !== segments.length - 1
-                        }
-                        show={<StrategySeparator text='AND' />}
-                    />
-                </Fragment>
-            ))}
-        </>
+        <SegmentItem
+            segment={segment}
+            constraintList={constraintList}
+            isExpanded
+        />
```

Comment:
Maybe constraintList should be passed as "children" prop or memoized, to save on re-renders.